modules/mono/glue/Managed/Files/StringExtensions.cs

```diff
@@ -97,6 +97,66 @@ public static string[] Bigrams(this string instance)
             return b;
         }
 
+        // <summary>
+        // Return the amount of substrings in string.
+        // </summary>
+        public static int Count(this string instance, string what, bool ignoreCase = false, int from = 0, int to = 0)
```

Comment:
Avoid double negatives. Instead of "ignoreCase", it should be "caseSensitive". Do a Ctrl+F in this file for `bool caseSensitive = true` to see examples of this.

Reply:
It's pretty common for case-sensitive-by-default APIs to have a flag for ignore case, not for enable case sensitivity (which is default). See e.g. any kind of grepping tool with `-i` flag.

`ignoreCase = false` is not a double negative, `dontIgnoreCase = true` would be.

That being said, if other extensions in this case favor `caseSensitive = true`, I'm not against that.

Reply:
Well, I changed to caseSensitive because the other functions in that file is using it

Reply:
IMO "ignore" is itself a negative, because it means "don't care about it", so `ignoreCase = false` would be "don't not care about it".

In C#, we specify in the method declaration which one is the default with true or false, so we don't need to depend on a flag existing or not - whichever makes the most readable code is best.

However, if this was changed to `ignoreCase`, it should be done for all of the other methods too.